js/privatebin.js

```diff
@@ -278,14 +276,19 @@ jQuery.PrivateBin = function($, sjcl, Base64, RawDeflate) {
                 return baseUri;
             }
 
-            // get official base uri string, from base tag in head of HTML
-            baseUri = document.baseURI;
+            // window.baseURI isn't emulated by JSdom
+            var loc = window.location;
+            baseUri = loc.href.substring(
+                0,
+                loc.href.length - loc.search.length - loc.hash.length
+            );
 
             // if base uri contains query string (when no base tag is present),
             // it is unwanted
-            if (baseUri.indexOf('?')) {
+            var queryIndex = baseUri.indexOf('?');
+            if (queryIndex !== -1) {
                 // so we built our own baseuri
-                baseUri = window.location.origin + window.location.pathname;
```

Comment:
Why does not `window.location.origin + window.location.pathname` work? Or what's bad about it?